mono/metadata/threads.c

```diff
@@ -1434,19 +1445,30 @@ ves_icall_System_Threading_Thread_GetCurrentThread (MonoError *error)
 	return MONO_HANDLE_NEW (MonoThreadObject, mono_thread_current ());
 }
 
+static MonoInternalThread*
+thread_handle_to_internal_ptr (MonoThreadObjectHandle thread_handle)
+{
+	return MONO_HANDLE_GETVAL(thread_handle, internal_thread); // InternalThreads are always pinned.
+}
+
 MonoBoolean
-ves_icall_System_Threading_Thread_Thread_internal (MonoThread *this_obj,
-												   MonoObject *start)
+ves_icall_System_Threading_Thread_Thread_internal (MonoThreadObjectHandle thread_handle, MonoObjectHandle start_handle, MonoError *error)
```

Comment:
lower in teh body of this icall we use `mono_set_pending_exception (mono_get_exception_thread_state (...))` - we should convert it to `mono_error_set_generic (error, "System.Threading", "ThreadStateException", msg)` (or even better - a new `mono_error_set_thread_state` that calls `mono_error_set_generic`).

Reply:
Let me look into that. I was following the pattern, but perhaps not a good pattern.